src/goto-programs/mark_decl_as_non_det.cpp

```diff
@@ -0,0 +1,41 @@
+#include <goto-programs/mark_decl_as_non_det.h>
+
+bool mark_decl_as_non_det::runOnFunction(
+  std::pair<const dstring, goto_functiont> &F)
+{
+  if(!F.second.body_available)
+    return false; // Didn't changed anything
+
+  for(auto it = F.second.body.instructions.begin();
+      it != F.second.body.instructions.end();
+      it++)
+  {
+    if(!it->is_decl())
+      continue;
+
+    auto decl = to_code_decl2t(it->code);
+    symbolt *s = context.find_symbol(decl.value);
+
+    // Global variables and function declaration shouldn`t reach here
+    assert(!s->static_lifetime || !s->type.is_code());
+
+    // Is the value initialized?
+    if(s->value.is_nil())
+    {
+      // Initialize it with nondet then
+      exprt nondet = exprt("sideeffect", s->type);
+      nondet.statement("nondet");
+      code_assignt assign(symbol_exprt(decl.value), nondet);
+      assign.location() = it->location;
```

Comment:
you can create an expr2tc here, instead of the old irep, so you don't have to call the migrate_expr method. An example can be found in `goto_k_inductiont::make_nondet_assign`.

Reply:
I've added a `gen_nondet` function